SonarTfsAnnotate/HistoryProvider.cs

```diff
@@ -44,25 +45,34 @@ public HistoryProvider(IEnumerable<Changeset> changesets)
 
         public bool Next()
         {
-            if (current - 1 >= 0)
+            while (true)
             {
-                Dispose(current - 1);
-            }
+                if (current - 1 >= 0)
+                {
+                    Dispose(current - 1);
+                }
 
-            current++;
-            if (current >= changesets.Count)
-            {
-                return false;
-            }
+                current++;
+                if (current >= changesets.Count)
+                {
+                    return false;
+                }
 
-            if (current + PREFETCH_SIZE < changesets.Count)
-            {
-                Prefetch(current + PREFETCH_SIZE);
-            }
+                if (current + PREFETCH_SIZE < changesets.Count)
+                {
+                    Prefetch(current + PREFETCH_SIZE);
+                }
+
+                manualResetEvents[current].WaitOne();
 
-            manualResetEvents[current].WaitOne();
+                if (!File.Exists(filenames[current]))
+                {
+                    // The download was not successful. Move on to the next file.
+                    continue;
```

Comment:
When we enter the next iteration in the while loop, the current count is incremented, thus we gloss over downloading a particular revision of a file. In such a case, the annotated value returned maybe incorrect for the file.

Ideally, a download failure should cause the annotation to fail for the file rather than having incorrect annotation.
Also, what is the exception you were getting?